persistence/trees/trees_test.go

```diff
@@ -1,19 +1,216 @@
+	"github.com/pokt-network/pocket/runtime"
+	"github.com/pokt-network/pocket/runtime/configs"
+	"github.com/pokt-network/pocket/runtime/test_artifacts"
+	"github.com/pokt-network/pocket/runtime/test_artifacts/keygen"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/messaging"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	DefaultChains = []string{"0001"}
+
+	DefaultDeltaBig   = big.NewInt(100)
+	DefaultAccountBig = big.NewInt(1000000)
+	DefaultStakeBig   = big.NewInt(1000000000000000)
+
+	DefaultAccountAmount = utils.BigIntToString(DefaultAccountBig)
+	DefaultStake         = utils.BigIntToString(DefaultStakeBig)
+
+	DefaultStakeStatus     = int32(coreTypes.StakeStatus_Staked)
+	DefaultPauseHeight     = int64(-1) // pauseHeight=-1 implies not paused
+	DefaultUnstakingHeight = int64(-1) // unstakingHeight=-1 implies not unstaking
+
+	testSchema = "test_schema"
+
+	genesisStateNumValidators   = 5
+	genesisStateNumServicers    = 1
+	genesisStateNumApplications = 1
+)
 
-// TECHDEBT(#836): Tests added in https://github.com/pokt-network/pocket/pull/836
 func TestTreeStore_Update(t *testing.T) {
-	// TODO: Write test case for the Update method
-	t.Skip("TODO: Write test case for Update method")
+	pool, resource, dbUrl := test_artifacts.SetupPostgresDocker()
+	t.Cleanup(func() {
+		if err := pool.Purge(resource); err != nil {
+			log.Fatalf("could not purge resource: %s", err)
+		}
+	})
+
+	t.Run("should update actor tree and commit", func(t *testing.T) {
+		pmod := newTestPersistenceModule(t, dbUrl)
+		context := NewTestPostgresContext(t, 0, pmod)
+
+		actor, err := createAndInsertDefaultTestApp(t, context)
+		assert.NoError(t, err)
+
+		t.Logf("actor inserted %+v", actor)
+		hash, err := context.ComputeStateHash()
+		assert.NoError(t, err)
+		assert.NotEmpty(t, hash)
+	})
+}
+
+func newTestPersistenceModule(t *testing.T, databaseUrl string) modules.PersistenceModule {
+	t.Helper()
+	teardownDeterministicKeygen := keygen.GetInstance().SetSeed(42)
+	defer teardownDeterministicKeygen()
+
+	cfg := &configs.Config{
+		Persistence: &configs.PersistenceConfig{
+			PostgresUrl:       databaseUrl,
+			NodeSchema:        testSchema,
+			BlockStorePath:    ":memory:",
+			TxIndexerPath:     ":memory:",
+			TreesStoreDir:     ":memory:",
+			MaxConnsCount:     5,
+			MinConnsCount:     1,
+			MaxConnLifetime:   "5m",
+			MaxConnIdleTime:   "1m",
+			HealthCheckPeriod: "30s",
+		},
+	}
+
+	genesisState, _ := test_artifacts.NewGenesisState(
+		genesisStateNumValidators,
+		genesisStateNumServicers,
+		genesisStateNumApplications,
+		genesisStateNumServicers,
+	)
+
+	runtimeMgr := runtime.NewManager(cfg, genesisState)
+
+	bus, err := runtime.CreateBus(runtimeMgr)
+	assert.NoError(t, err)
```

Comment:
Is there a reason for `assert` over `require`?

AFAIK `assert` fails but keeps the test running, whereas `require` panics on the error.